test/functional/testlib.bash

```diff
@@ -3167,98 +3219,125 @@ list_tests() { # swupd_function
 # tests since all tests require at least the creation of a  test environment
 #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
-global_setup_used=true
-global_env=false
-
 setup() {
 
-	print "\\n$sep"
+	print "\\n\\n\\n$sep"
 	print "$alt_sep"
-	print "$sep"
-	# a global environment will always use number 1, so check for that
-	# environment first
-	TEST_NAME_SHORT="$TEST_NAME"
-	if [ -d "$TEST_NAME"_1 ]; then
-		# if a global environment is found we want to continue using that
-		# as our test environment for the rest of the tests, if not continue
-		# numbering the envs according to the test number
-		if [ -e "$TEST_NAME"_1/.global_env ]; then
-			TEST_NAME="$TEST_NAME"_1
-			global_env=true
-		else
-			TEST_NAME="$TEST_NAME"_"$BATS_TEST_NUMBER"
+	print "$sep\\n"
+
+	debug_msg "Test file: $TEST_NAME.bats"
+	debug_msg "BATS test number: $BATS_TEST_NUMBER"
+
+	# run the global_setup only once
+	if [ "$BATS_TEST_NAME" = "${BATS_TEST_NAMES[0]}" ]; then
+
+		# if a global environment exists at this point it means
+		# it was a leftover from a previous execution, remove it
+		if [ -d "$TEST_NAME" ]; then
+			print "\\nAn old test environment was found for this test: $TEST_NAME"
```

Comment:
Should we change this to debug as well? Not really relevant unless we run with DEBUG_TETS

Reply:
yeah, we can make that a debug message now